modin/core/execution/client/container.py

```diff
@@ -0,0 +1,743 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module contains ``ForwardingQueryCompilerContainer`` class."""
+
+import numpy as np
+import pandas
+from typing import Any, NamedTuple, Optional, Union
+from uuid import UUID, uuid4
+
+from modin.core.storage_formats.base.query_compiler import BaseQueryCompiler
+
+
```

Comment:
Is it supposed to have multiple, concurrent clients? If so, then, this class seems not thread safe.

Reply:
Modin users are supposed to only have one client.